src/application-navigation/ApplicationNavigation.jsx

```diff
@@ -2,17 +2,18 @@ import React, {
   useRef, useCallback, Suspense,
 } from 'react';
 import PropTypes from 'prop-types';
-import TerraApplicationNavigation from 'terra-application-navigation';
-import {
-  titleConfigPropType, navigationItemsPropType, extensionItemsPropType, utilityItemsPropType, userConfigPropType,
-} from 'terra-application-navigation/lib/utils/propTypes';
 
 import ApplicationErrorBoundary from '../application-error-boundary';
 import ApplicationLoadingOverlay, { ApplicationLoadingOverlayProvider } from '../application-loading-overlay';
 import { ApplicationStatusOverlayProvider } from '../application-status-overlay';
 import { NavigationPromptCheckpoint, navigationPromptResolutionOptionsShape, getUnsavedChangesPromptOptions } from '../navigation-prompt';
 import { ApplicationIntlContext } from '../application-intl';
 
+import TerraApplicationNavigation from './terra-application-navigation/ApplicationNavigation';
+import {
+  titleConfigPropType, navigationItemsPropType, extensionItemsPropType, utilityItemsPropType, userConfigPropType,
+} from './terra-application-navigation/utils/propTypes';
```

Comment:
Should the terra namespace be dropped? Looking at the styles, they are weird seeing `terra-application-application`

I was also wondering if the structure should be flattened to move the terra-application-navigation contents up a dir level?
```
src/
    application-navigation/
        ApplicationNavigation.jsx
        index.js
        NavigationLayout (instead of terra-application-navigation/ApplicationNavigation)
```

Reply:
I will move things around quite a bit with the introduction of the ApplicationContainer in my next PR. I wanted to get this code landed here first because there's so much of it. A bit of a half-step I know, but I think it'll make more sense to reorganize this in a followup.

I believe we need the `--terra-application` prefix to align with the naming rules, but I agree that seeing application twice in there is awkward. I can update these again when I come up with a better name for this thing.